File: src/main/java/com/legendary/coffeeShop/dao/entities/Order.java
package com.legendary.coffeeShop.dao.entities;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.sql.Timestamp;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@Table(name = "orders")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_orders", nullable = false)
    private User user;

    @OneToMany(mappedBy = "id", cascade = CascadeType.ALL)
    private List<OrderItem> orderItems;

    @Column(name = "creation_time")
    private Timestamp creationTime;

    @Column(name = "update_time")
    private Timestamp updateTime;

    @Column(name = "order_status")
    private OrderStatus orderStatus;

}

File: public/src/app/models/cart/order.ts
import {OrderStatus} from '@models/cart/order-status.enum';
import {OrderItem} from '@models/cart/order-item';


export interface Order {
  id?: number;
  orderStatus: OrderStatus;
  updateTime: string;
  creationTime: string;
  orderItems: OrderItem[];
  price?: number;
  displayDate?: string;
  orderItemProducts?: string[];
}
